hydroflow/src/compiled/pull.rs

```diff
@@ -19,6 +20,28 @@ impl<K, V1, V2> Default for JoinState<K, V1, V2> {
     }
 }
 
+enum Side {
+    Left,
+    Right,
+}
+pub struct CrossJoinState<V1, V2> {
+    ltab: Vec<V1>,
+    rtab: Vec<V2>,
+    eventside: Side,
+    opposite_ix: Range<usize>,
+}
+
+impl<'a, V1, V2> Default for CrossJoinState<V1, V2> {
+    fn default() -> Self {
+        Self {
+            ltab: Vec::new(),
+            rtab: Vec::new(),
+            eventside: Side::Left,
+            opposite_ix: 0..0,
```

Comment:
the upper bound on this will always be the length of the opposite side right? so we could just track the index of the opposite side we're currently in? or even just store it in the enum maybe:
```rust
enum IterationState {
    Left(usize),
    Right(usize),
}
```

You could maybe go a bit further and encode the state a bit more explicitly by including the current row,

```rust
enum IterationState<V1, V2> {
    Idle,
    Left(V1, usize),
    Right(V2, usize),
}
```
which has a couple benefits:
1. you don't have to `unwrap` the last row of the corresponding side, since it will be guaranteed to be present in the enum, and
2. rather than relying on `return` to dictate whether or not to fall through to fetching a new row, you can make `next` one big loop:

```rust
loop {
  match self.iteration_state {
    IterationState::Left(...) => // emit the next left row, or update the state to Idle and append the row to ltab
    IterationState::Right(...) => // emit the next right row, or update the state to Idle and append the row to rtab
    IterationState::Idle => // fetch a new row and update the state
  }
}
```